arkouda-docker/build_docker_image.py

```diff
@@ -0,0 +1,196 @@
+                                               '--build-arg', f'ARKOUDA_BRANCH_NAME={distro}',
+                                               '-f',file,
+                                               '-t', dockerTag, '.'], stdout=subprocess.DEVNULL)
+        print(result)
+
+    def buildArkoudaUdpServer(dockerRepo: str, chapelVersion: str, file: str, dockerTag: str, distro: str, tag: Optional[str]) -> None:
+        result = subprocess.run(args=['docker','build',
+                                               '--build-arg', f'CHAPEL_UDP_IMAGE={generateChplUdpVersion(chapelVersion)}',
+                                               '--build-arg', f'ARKOUDA_DISTRO_NAME={getDistroName(distro=distro, tag=tag)}',
+                                               '--build-arg', f'ARKOUDA_DOWNLOAD_URL={generateArkoudaDownloadUrl(tag=tag,branch=distro)}',
+                                               '--build-arg', f'ARKOUDA_BRANCH_NAME={distro}',
+                                               '--build-arg', 'ARKOUDA_INTEGRATION_DOWNLOAD_URL=https://github.com/Bears-R-Us/arkouda-contrib/archive/refs/heads/main.zip',
+                                               '--build-arg', 'ARKOUDA_INTEGRATION_DISTRO_NAME=main',
+                                               '-f',file,
+                                               '-t', dockerTag, '.'], stdout=subprocess.DEVNULL)
+        print(result)
+
+    def buildChapelUdp(dockerRepo: str, chapelVersion: str, file: str, dockerTag: str) -> None:
+        result = subprocess.run(args=['docker','build',
+                                               '--build-arg', f'CHPL_BASE_IMAGE=ubuntu:22.04',
+                                               '--build-arg', f'CHPL_VERSION={chapelVersion}',
+                                               '--build-arg', f'CHPL_UDP_IMAGE_REPO={dockerRepo}',
+                                               '-f',file,
+                                               '-t', f'{dockerRepo}/{file}:{chapelVersion}', '.'], stdout=subprocess.DEVNULL)
+        print(result)
+
+
+    dockerTag = generateBuildTag(dockerRepo=dockerRepo, file=file, tag=tag,distro=distro)
+
+    if file == 'arkouda-full-stack':   
+        buildArkoudaFullStack(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag,distro=distro,tag=tag)
+    elif file == 'arkouda-smp-server':
+        buildArkoudaSmpServer(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag,distro=distro,tag=tag)
+    elif file == 'arkouda-udp-server':
+        buildArkoudaUdpServer(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag,distro=distro,tag=tag)
+    else:
+        buildChapelUdp(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag)
+
+def getDistroName(distro: str, tag: Optional[str]) -> str:
+    if tag:
+        return tag
+    else:
+        return distro
```

Comment:
this is just stylstic, but i like to do a ternary in cases like this 
```python
return tag if tag else distro
```

Reply:
@pierce314159 cool, good idea

Reply:
@pierce314159 made the change, please check it out.